src/sockperf.cpp

```diff
@@ -725,19 +831,48 @@ static int proc_mode_ping_pong(int id, int argc, const char **argv) {
 
     /* Set command line specific values */
     if (!rc && self_obj) {
-        if (!rc && aopt_check(self_obj, 't')) {
-            const char *optarg = aopt_value(self_obj, 't');
-            if (optarg) {
-                errno = 0;
-                int value = strtol(optarg, NULL, 0);
-                if (errno != 0 || value <= 0 || value > MAX_DURATION) {
-                    log_msg("'-%c' Invalid duration: %s", 't', optarg);
+        if (!rc && aopt_check(self_obj, 'n')) {
+            if (!aopt_check(self_obj, 't') && !aopt_check(self_obj, OPT_MPS)) {
+                const char *optarg = aopt_value(self_obj, 'n');
+                if (optarg) {
+                    errno = 0;
+                    int value = strtol(optarg, NULL, 0);
+                    if (errno != 0 || value <= 0 || value > MAX_PACKET_NUMBER) {
+                        log_msg("'-%c' Invalid number of packets: %s", 'n', optarg);
+                        rc = SOCKPERF_ERR_BAD_ARGUMENT;
+                    } else {
+                        s_user_params.measurement = NUMBER_BASED;
+                        s_user_params.number_test_target = value;
+                    }
+                } else {
+                    log_msg("'-%c' Invalid value", 'n');
                     rc = SOCKPERF_ERR_BAD_ARGUMENT;
+                }
+            } else {
+                log_msg("-n conflicts with -t,--mps options");
+                rc = SOCKPERF_ERR_BAD_ARGUMENT;
+            }
+        }
+
+        if (!rc && aopt_check(self_obj, 't')) {
+            if (!aopt_check(self_obj, 'n')) {
+                const char *optarg = aopt_value(self_obj, 't');
+                if (optarg) {
+                    errno = 0;
+                    int value = strtol(optarg, NULL, 0);
+                    if (errno != 0 || value <= 0 || value > MAX_DURATION) {
+                        log_msg("'-%c' Invalid duration: %s", 't', optarg);
+                        rc = SOCKPERF_ERR_BAD_ARGUMENT;
+                    } else {
+                        s_user_params.measurement = TIME_BASED;
+                        s_user_params.sec_test_duration = value;
+                    }
                 } else {
-                    s_user_params.sec_test_duration = value;
+                    log_msg("'-%c' Invalid value", 't');
+                    rc = SOCKPERF_ERR_BAD_ARGUMENT;
                 }
             } else {
-                log_msg("'-%c' Invalid value", 't');
+                log_msg("-t conflicts with -n option");
```

Comment:
The conflict between `-t` and `-n` has already been checked in previous `if` (lines 834-855).